rest/src/main/java/de/starwit/rest/controller/AppTemplateController.java

```diff
@@ -51,14 +51,14 @@ public AppTemplate findById(@PathVariable("templateId") Long id) {
     @IsAdmin
     @Operation(summary = "Create appTemplate (location, branch, description, credentialsRequired)")
     @PostMapping
-    public AppTemplate save(@Valid @RequestBody SaveAppTemplateDto appTemplateDto) {
-        return update(appTemplateDto);
+    public AppTemplate save(@Valid @RequestBody SaveAppTemplateDto appTemplateDto, Principal principal) {
+        return update(appTemplateDto, principal);
     }
 
     @IsAdmin
     @Operation(summary = "Update appTemplate (location, branch, description, credentialsRequired)")
     @PutMapping
-    public AppTemplate update(@Valid @RequestBody SaveAppTemplateDto appTemplateDto) {
+    public AppTemplate update(@Valid @RequestBody SaveAppTemplateDto appTemplateDto, Principal principal) {
```

Comment:
Finde das ein wenig unschön, den Principle so herunterzureichen. Eventuell könnte man im UserService was anlegen um den Pricipal auszulesen: https://www.baeldung.com/get-user-in-spring-security#interface

Reply:
Und in ein Dto oder Entität packen....

Reply:
Das wird nicht heruntergereicht, sondern vom Framework aufgelöst. Im Link steht das als eine Lösung drin.